opengrok-indexer/src/test/java/org/opengrok/indexer/analysis/ocaml/OCamlXrefTest.java

```diff
@@ -0,0 +1,179 @@
+                + "href=\"http://localhost:8080/source/default/style.css\" /><title>OCaml Xref Test</title></head>");
+        os.println("<body><div id=\"src\"><pre>");
+        Writer w = new StringWriter();
+        OCamlAnalyzerFactory fac = new OCamlAnalyzerFactory();
+        AbstractAnalyzer analyzer = fac.getAnalyzer();
+        WriteXrefArgs args = new WriteXrefArgs(new InputStreamReader(is, StandardCharsets.UTF_8), w);
+        args.setDefs(defs);
+        Xrefer xref = analyzer.writeXref(args);
+        os.print(w.toString());
+        os.println("</pre></div></body></html>");
+        return xref.getLOC();
+    }
+
+    @Test
+    void sampleTest() throws IOException {
+        // load sample source
+        InputStream sampleInputStream = getClass().getClassLoader().getResourceAsStream(
+                "analysis/ocaml/sample.ml");
+        ByteArrayOutputStream sampleOutputStream = new ByteArrayOutputStream();
+
+        Definitions defs = new Definitions();
+        defs.addTag(6, "x'y'", "functions",
+            "let x'y' = let f' = 1; g'h = 2 in f' + g'h", 0, 0);
+
+        int actLOC;
+        try {
+            actLOC = writeOCamlXref(sampleInputStream, new PrintStream(sampleOutputStream), defs);
+        } finally {
+            sampleInputStream.close();
+            sampleOutputStream.close();
+        }
+
+        // load expected xref
+        InputStream expectedInputStream = getClass().getClassLoader().getResourceAsStream(
+                "analysis/ocaml/sample_xref.html");
+        ByteArrayOutputStream expectedOutputSteam = new ByteArrayOutputStream();
+        try {
+            byte[] buffer = new byte[8192];
+            int numBytesRead;
+            do {
+                numBytesRead = expectedInputStream.read(buffer, 0, buffer.length);
+                if (numBytesRead > 0) {
+                    expectedOutputSteam.write(buffer, 0, numBytesRead);
+                }
+            } while (numBytesRead >= 0);
+        } finally {
+            expectedInputStream.close();
+            expectedOutputSteam.close();
+        }
+
+        String[] actual = new String(sampleOutputStream.toByteArray(), StandardCharsets.UTF_8).split("\\r?\\n");
+        String[] expected = new String(expectedOutputSteam.toByteArray(), StandardCharsets.UTF_8).split("\\r?\\n");
+        assertLinesEqual("OCaml sampleTest()", expected, actual);
+        assertEquals(17, actLOC, "OCaml LOC");
+    }
+
+    @Test
+    void sampleTest2() throws IOException {
+        writeAndCompare("analysis/ocaml/sample2.ml",
+            "analysis/ocaml/sample2_xref.html",
+            getTagsDefinitions(), 11);
+    }
+
+    private void writeAndCompare(String sourceResource, String resultResource,
+        Definitions defs, int expLOC) throws IOException {
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        InputStream res = getClass().getClassLoader().getResourceAsStream(
```

Comment:
feel free to join these lines, the lines can be pretty long https://github.com/oracle/opengrok/blob/1c311b09924319e3b7e7d45417e700110845363e/dev/checkstyle/style.xml#L24